test/unit/http/request_test.dart

```diff
@@ -312,7 +299,12 @@ void _runCommonRequestSuiteFor(String name,
       await Future.delayed(Duration(milliseconds: 100));
       request.abort();
       expect(future, throwsA(isA<transport.RequestException>()));
-      await future.catchError((_) {});
+      try {
+        await future;
+        fail('Expected RequestException, but future completed successfully.');
+      } catch (error) {
+        expect(error, isA<transport.RequestException>());
+      }
```

Comment:
This works, but seems to duplicate the assertion just above where we are also asserting that it throws. Do we have to do this instead of using `.catchError` because of non-nullability?